Tests/CommandsTests/PackageToolTests.swift

```diff
@@ -529,4 +530,60 @@ final class PackageToolTests: XCTestCase {
                 """)
         }
     }
+
+    func testMirrorConfig() {
+        mktmpdir { prefix in
+            let fs = localFileSystem
+            let packageRoot = prefix.appending(component: "Foo")
+            let configFile = packageRoot.appending(components: ".swiftpm", "config")
+
+            fs.createEmptyFiles(at: packageRoot, files:
+                "/Sources/Foo/Foo.swift",
+                "/Tests/FooTests/FooTests.swift",
+                "/Package.swift",
+                "anchor"
+            )
+
+            // Test writing.
+            try execute(["config", "set-mirror", "--package-url", "https://github.com/foo/bar", "--mirror-url", "https://mygithub.com/foo/bar"], packagePath: packageRoot)
+            try execute(["config", "set-mirror", "--package-url", "git@github.com:apple/swift-package-manager.git", "--mirror-url", "git@mygithub.com:foo/swift-package-manager.git"], packagePath: packageRoot)
+            XCTAssertTrue(fs.isFile(configFile))
+
+            // Test reading.
+            XCTAssertEqual(try execute(["config", "get-mirror", "--package-url", "https://github.com/foo/bar"], packagePath: packageRoot),
+                "https://mygithub.com/foo/bar")
+            XCTAssertEqual(try execute(["config", "get-mirror", "--package-url", "git@github.com:apple/swift-package-manager.git"], packagePath: packageRoot),
+                "git@mygithub.com:foo/swift-package-manager.git")
+
+            func check(stderr: String, _ block: () throws -> ()) {
+                do {
+                    try block()
+                    XCTFail()
+                } catch SwiftPMProductError.executionFailure(_, _, let stderrOutput) {
+                    XCTAssertEqual(stderrOutput, stderr)
+                } catch {
+                    XCTFail("unexpected error: \(error)")
+                }
+            }
+
+            check(stderr: "not found\n") {
+                try execute(["config", "get-mirror", "--package-url", "foo"], packagePath: packageRoot)
+            }
+
+            // Test deletion.
+            try execute(["config", "unset-mirror", "--package-url", "https://github.com/foo/bar"], packagePath: packageRoot)
+            try execute(["config", "unset-mirror", "--mirror-url", "git@mygithub.com:foo/swift-package-manager.git"], packagePath: packageRoot)
```

Comment:
The proposal mentions to implement `unset-mirro --all` but its not yet implemented. 

## Proposal
https://github.com/apple/swift-evolution/blob/master/proposals/0219-package-manager-dependency-mirroring.md#dependency-mirroring 

## Usacase 
<img width="600" alt="screenshot 2018-10-14 at 07 56 39" src="https://user-images.githubusercontent.com/4518004/46913086-e0ddce00-cf86-11e8-8bb6-4940b0b3d3a7.png">

## what next?
This can be done in two ways at least from my naive eye:
1. use the options parses as all of `get-mirror` and `set-mirror` or `unset-mirror` is done. Internally ask the json from file and remove each one.
2.  Parse and then remove the entire json object from the file at once. `.swiftpm/config`

Any thoughts?

Reply:
Given that we currently only store the mirror information in the config file, I don't see a need to implement the `--all` flag. The user can just delete config file by hand for now. We can add the flag in the future when we start adding more stuff in the config file.